src/Request.cpp

```diff
@@ -73,13 +79,13 @@ auto Request::SetOnCompleteHandler(
     m_on_complete_handler = std::move(on_complete_handler);
 }
 
-auto Request::SetUrl(const std::string& url) -> bool
+auto Request::SetUrl(std::string_view url) -> bool
 {
     if (url.empty()) {
         return false;
     }
 
-    auto error_code = curl_easy_setopt(m_curl_handle, CURLOPT_URL, url.c_str());
+    auto error_code = curl_easy_setopt(m_curl_handle, CURLOPT_URL, url.data());
```

Comment:
dangerzone, url.data() isn't guaranteed to be null terminated because it's a string_view, so that's likely why this wasn't a string_view before